atomate/vasp/firetasks/electrode_tasks.py

```diff
@@ -0,0 +1,311 @@
+        cia.get_labels()
+
+        insert_sites = []
+        seent = set()
+
+        cia._extrema_df.sort_values(by=["avg_charge_den"], inplace=True)
+        for itr, li_site in cia._extrema_df.iterrows():
+            if len(insert_sites) >= max_insertions:
+                break
+            li_site = cia._extrema_df.iloc[itr]
+            lab = li_site["site_label"]
+            if lab not in seent:
+                insert_sites.append([li_site["a"], li_site["b"], li_site["c"]])
+                seent.add(lab)
+
+        logger.info(
+            f"Found {len(insert_sites)} insertion sites for task : {base_task_id}"
+        )
+
+        # Since we are only analyzing a single charge density
+        # The we will only perform one set of atoms insertions at a given time
+        # Thus, we just have to update the global fw_spec
+        # (no need to pass this information from fw to fw)
+        return FWAction(
+            update_spec={
+                "insert_sites": insert_sites,
+                "base_task_id": base_task_id,
+                "base_structure": chgcar.structure.as_dict(),
+            }
+        )
+
+
+@explicit_serialize
+class GetInsertionCalcs(FiretaskBase):
+    """
+    Using a list of insertion sites and a base structure, get a new set of calculations
+
+    fw_spec requires:
+    - "base_task_id" : the task_id for the charge density that is being analyzed
+    - "base_structure": the base structure that the ion will be inserted on
+    - "insert_sites": The list of insertion sites
+    - "working_ion": The name of the ion to be inserted
+
+    Optional fw_spec:
+    - "allow_fizzled_parents": If True, perform the static calculation and charge density
+        analysis as long as a single optimization is finished. Default is False
+    - "optimizefw_kwargs":  kwargs for the OptimizeFW created by the present task
+    """
+
+    def run_task(self, fw_spec):
+
+        insert_sites = fw_spec.get("insert_sites", [])
+        base_task_id = fw_spec.get("base_task_id")
+        base_structure = Structure.from_dict(fw_spec.get("base_structure"))
+        working_ion = fw_spec.get("working_ion")
+        allow_fizzled_parents = fw_spec.get("allow_fizzled_parents", False)
+        optimizefw_kwargs = fw_spec.get("optimizefw_kwargs", {})
+        n_ion = int(base_structure.composition.element_composition[working_ion]) + 1
+
+        new_fws = []
+        for itr, isite in enumerate(insert_sites):
+            inserted_structure = base_structure.copy()
+            fpos = isite
+            inserted_structure.insert(0, working_ion, fpos, properties={"magmom": 1.0})
+            additional_fields = {"insertion_fpos": fpos, "base_task_id": base_task_id}
+
+            # Create new fw
+            fw = OptimizeFW(
+                inserted_structure,
+                name=f"structure optimization-{itr}",
+                override_default_vasp_params={
+                    "user_incar_settings": {"NSW": 9999}
```

Comment:
😨 This is an enormous number. Do the insertions really need 10,000 steps? I'm worried that if VASP can't optimise the structure, it will burn computer time with the atoms bouncing around without converging.

Reply:
oop I think that was introduced during testing. I have set it to a more reasonable value.